src/blockencodings.cpp

```diff
@@ -130,6 +131,35 @@ ReadStatus PartiallyDownloadedBlock::InitData(const CBlockHeaderAndShortTxIDs& c
         if (mempool_count == shorttxids.size())
             break;
     }
+    }
```

Comment:
This isnt true? This is only true if the witness hash (and thus, the txn) are identical.